pkg/proxy/userspace/proxier.go

```diff
@@ -687,12 +696,16 @@ func iptablesCommonPortalArgs(destIP net.IP, destPort int, protocol api.Protocol
 		args = append(args, "-d", fmt.Sprintf("%s/32", destIP.String()))
 	}
 
+	if excludeContainerSourceTraffic {
+		args = append(args, "-m", "physdev", "--physdev-is-in")
```

Comment:
Ahh, I see you have named a variable backwards.  This flag is called "exclude" but you pass false to it when you want to exclude.  Try it the other way.

You only care about whether it is a bridged packet IFF the caller said to exclude.

clusterIP: don't care about bridging
load-balancer: don't care about bridging
external IP: don't allow bridging

I think that handles what you want here.